vms/platformvm/validators/manager_test.go

```diff
@@ -0,0 +1,445 @@
+// Copyright (C) 2019-2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package validators
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/avalanchego/chains"
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/snow/choices"
+	"github.com/ava-labs/avalanchego/snow/uptime"
+	"github.com/ava-labs/avalanchego/snow/validators"
+	"github.com/ava-labs/avalanchego/utils/constants"
+	"github.com/ava-labs/avalanchego/utils/crypto/bls"
+	"github.com/ava-labs/avalanchego/utils/timer/mockable"
+	"github.com/ava-labs/avalanchego/utils/units"
+	"github.com/ava-labs/avalanchego/vms/platformvm/blocks"
+	"github.com/ava-labs/avalanchego/vms/platformvm/config"
+	"github.com/ava-labs/avalanchego/vms/platformvm/metrics"
+	"github.com/ava-labs/avalanchego/vms/platformvm/reward"
+	"github.com/ava-labs/avalanchego/vms/platformvm/state"
+)
+
+// AVAX asset ID in tests
+var defaultRewardConfig = reward.Config{
+	MaxConsumptionRate: .12 * reward.PercentDenominator,
+	MinConsumptionRate: .10 * reward.PercentDenominator,
+	MintingPeriod:      365 * 24 * time.Hour,
+	SupplyCap:          720 * units.MegaAvax,
+}
+
+func TestVM_GetValidatorSet(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	// Populate the validator set to use below
+	var (
+		numVdrs       = 4
+		vdrBaseWeight = uint64(1_000)
+		vdrs          []*validators.Validator
+	)
+
+	for i := 0; i < numVdrs; i++ {
+		sk, err := bls.NewSecretKey()
+		require.NoError(t, err)
+
+		vdrs = append(vdrs, &validators.Validator{
+			NodeID:    ids.GenerateTestNodeID(),
+			PublicKey: bls.PublicFromSecretKey(sk),
+			Weight:    vdrBaseWeight + uint64(i),
+		})
+	}
+
+	type test struct {
+		name string
+		// Height we're getting the diff at
+		height             uint64
+		lastAcceptedHeight uint64
+		subnetID           ids.ID
+		// Validator sets at tip
+		currentPrimaryNetworkValidators []*validators.Validator
+		currentSubnetValidators         []*validators.Validator
+		// Diff at tip, block before tip, etc.
+		// This must have [height] - [lastAcceptedHeight] elements
```

Comment:
Did you mean?

```suggestion
		// This must have [lastAcceptedHeight] - [height] elements
```

(same below)

Reply:
Done